src/Type/Php/FilterVarDynamicReturnTypeExtension.php

```diff
@@ -222,4 +229,26 @@ private function getFlagsValue(Type $exprType): Type
 		return $exprType->getOffsetValueType($this->flagsString);
 	}
 
+	private function canStringBeSanitized(Type $filterType, int $filterValue, ?Node\Arg $flagsArg, Scope $scope): bool
+	{
+		if (!$filterType instanceof StringType) {
```

Comment:
`instanceof *Type` is rarely correct, see: https://phpstan.org/developing-extensions/type-system

For example this would fail for `numeric-string` or `non-empty-string`

Reply:
Thanks, fixed.

Reply:
Adding a test with e.g. a numeric-string would cover the last change

Reply:
I can, but as I was saying below, that wouldn't really accomplish anything. `$filterType` is the type returned by the filter, per `getFilterTypeMap()`, not the type of the input itself. I can still add a test if you think it's necessary.